extensions/pubsub/pubsub.ui.runtime.ts

```diff
@@ -0,0 +1,82 @@
+import { UIAspect, UIRuntime } from '@teambit/ui';
+
+import { connectToChild } from 'penpal';
+
+import { BitBaseEvent } from './bit-base-event';
+import { PubsubAspect } from './pubsub.aspect';
+
+export class PubsubUI {
+  private topicMap = {};
+  private _childs;
+
+  private getAllIframes = () => {
+    return Array.from(document.getElementsByTagName('iframe'));
+  };
+
+  private connectToIframe = async (iframe) => {
+    const self = this;
+
+    return await connectToChild({
+      timeout: 500,
+      iframe,
+      methods: {
+        sub(topicUUID, callback) {
+          return self.sub(topicUUID, callback);
+        },
+        pub(topicUUID, event: BitBaseEvent<any>) {
+          return self.pub(topicUUID, event);
+        },
+      },
+    })
+      .promise.then((child) => child)
+      .catch((err) => {
+        return this.connectToIframe(iframe);
+      });
+  };
+
+  private updateConnectionsList() {
+    const _iframes = this.getAllIframes();
+    return _iframes.map((iframe) => this.connectToIframe(iframe));
```

Comment:
please check if a disconnect is required

Reply:
'required' what do you mean?
required for what?

Reply:
I mean, if an iframe is removed, then we should call disconnect, no?